lib/vanagon/driver.rb

```diff
@@ -33,6 +35,19 @@ def initialize(platform, project, options = { :configdir => nil, :target => nil,
       raise Vanagon::Error.wrap(e, "Could not load the desired engine '#{engine}'")
     end
 
+    def filter_out_components(project, only_build)
+      filtered_components = []
+      only_build.each do |comp|
+        filtered_components += @project.filter_component(comp)
+      end
+      project.components = filtered_components.uniq
```

Comment:
Sorry, I gave you the final product of some of my tinkering, but I didn't give you the steps. The real chunk of code I should have thrown at you is:

```ruby
filtered_components = only_build.map { |comp| @project.filter_component(comp) }.uniq
```